src/components/modal/profile/index.tsx

```diff
@@ -6,14 +6,22 @@ import { InteractiveComponentsContext } from 'contexts/interactiveComponentsCont
 import CloseIcon from '@mui/icons-material/Close';
 import { useTheme } from '@mui/material/styles';
 import { Modal } from 'components/modal';
+import PATHS from 'constants/paths';
+import { useNavigate } from 'react-router-dom';
+import { AuthContext } from 'contexts/authContext';
+import { authApiService } from 'services/api/AuthApiService';
 
 export const ProfileModal = () => {
   const { isModalOpened, closeModal } = useContext(InteractiveComponentsContext);
   const theme = useTheme();
+  const navigate = useNavigate();
+  const { setUser } = useContext(AuthContext);
```

Comment:
Create inside the useAuthenticatedUser hook a logout function and use that. All the logic related to auth-user should be kept in that hook